spring-core/src/main/java/org/springframework/core/annotation/AnnotationTypeMapping.java

```diff
@@ -750,7 +758,7 @@ class MirrorSet {
 
 			private int size;
 
-			private final int[] indexes = new int[attributes.size()];
+			private final int[] indexes = attributes.size() > 0 ? new int[attributes.size()] : EMPTY_INT_ARRAY;
```

Comment:
@kilink Could you please double check if you have a use case where `attributes.size() == 0` at this line? I was not able to find one in Petclinic JDBC. If you can't, please revert that specific change (and keep other ones that looks legit).